src/sage/rings/semirings/tropical_polynomial.py

```diff
@@ -0,0 +1,724 @@
+            roots = sorted(self.roots())
+            if roots==[] or self.parent().base().zero() in roots:
+                return plot(f, xmin=-1, xmax=1)
+            else:
+                return plot(f, xmin=roots[0]-1, xmax=roots[-1]+1)
+        elif xmin is None or xmax is None:
+            raise ValueError(f"Expected 2 inputs for xmin and xmax, but got 1")
+        elif (xmin>=xmax):
+            raise ValueError(f"xmin = {xmin} should be less than xmax = {xmax}")
+        else:
+            return plot(f, xmin=xmin, xmax=xmax)
+
+
+class TropicalPolynomialSemiring(UniqueRepresentation, Parent):
+    """
+    Semiring structure of tropical polynomials in one variable
+    """
+
+    @staticmethod
+    def __classcall_private__(cls, base_semiring, names=None):
+        if names is None:
+           names = 'x'
+        if isinstance(names, str):
+            names = (names,)
+        return super().__classcall__(cls, base_semiring, tuple(names))
+
+    def __init__(self, base_semiring, names):
+        """
+        EXAMPLES:
+
+            sage: T = TropicalSemiring(QQ, use_min=False)
+            sage: R.<x> = PolynomialRing(T)
+            sage: x.parent()
+            Univariate Tropical Polynomial Semiring in x over Rational Field
+            sage: (x + T(1)*x^2) * R(3)
+            4*x^2 + 3*x
+            sage: category(R)
+            Category of semirings
+
+        """
+        if not isinstance(base_semiring, TropicalSemiring):
+            raise ValueError(f"{base_semiring} is not a tropical semiring")
+        Parent.__init__(self, base=base_semiring, names=names, category=Semirings())
+
+    Element = TropicalPolynomial
+
+    def _element_constructor_(self, x, check=True):
+        """
+        Convert ``x`` into this tropical polynomial semiring
+        """
+        C = self.element_class
+        if isinstance(x, (list, tuple)):
+            for i, coeff in enumerate(x):
+                if coeff == 0:
+                    x[i] = self.base()(0)
+        return C(self, x, check=check)
+
+    def _repr_(self):
+        return (f"Univariate Tropical Polynomial Semiring in {self.variable_name()}"
+            f" over {self.base_ring().base_ring()}")
+
+    @cached_method
+    def gen(self, n=0):
+        """
+        Return the indeterminate generator of this polynomial ring.
+        """
+        if n != 0:
+            raise IndexError("generator n not defined")
+        term = [0, self.base()(0)]
+        return self.element_class(self, term, is_gen=True)
+    
+    def gens(self):
+        """
+        Return a tuple whose entries are the generators for this
+        object, in order.
+        """
+        self._gens = tuple(self.gen(i) for i in range(self.ngens()))
+        return self._gens
+    
+    def ngens(self):
+        """
+        Return the number of generators of this polynomial ring, which is 1
+        since it is a univariate polynomial ring.
+        """
+        return 1
```

Comment:
```suggestion
        return ZZ.one()
```
Python `int`s should be avoided in public method output.